components/eam/src/chemistry/utils/read_spa_data.F90

```diff
@@ -0,0 +1,93 @@
+module read_spa_data
+
+!BSINGH - I didn't add any comments here due to time constrain but I will add comments later
+
+use tracer_data
+
+implicit none
+
+private
+
+public :: read_spa_data_init, read_spa_data_adv, read_spa_data_register
+
+type(trfld), pointer :: spa_fields(:)
+type(trfile)         :: spa_file
+logical              :: rmv_file = .false.
+
+!fields to read from the file
+character(len=16), parameter :: pbuf_names(59) = &
```

Comment:
May be split these apart and use these as in the radiation code and then combine them in one array. This will save us from writing these all over again in the radiation code and it will be easy to maintain.

Also, use a parameter variable to store `59`